### ISSUE TITLE
ExtensionArray.astype docstring

### ISSUE THREAD

**Issue by andrewgsavage** [pandas-dev/pandas#24773]
https://github.com/pandas-dev/pandas/blob/6d3565a5181b3d0fb8342e69dd1d21a98ac3d380/pandas/core/arrays/base.py#L324

Should the docstring say that this should return an ExtensionArray when an ExtensionType is passed into it?
I note that PeriodArray behaves like this. 
```
import pandas as pd
a=pd.core.arrays.period_array([2000, 2001, 2002], freq='D')
type(a.dtype)
```

pandas.core.dtypes.dtypes.PeriodDtype

```
type(a.astype(a.dtype))
```
pandas.core.arrays.period.PeriodArray

**Comment by TomAugspurger** [comment]
Yes, it should return an ExtensionArray when passed an ExtensionDtype.

This ends up making `.astype` quite complicated in practice. Pandas should provide some helpers here (though we had an issue for that, but couldn't quickly find one).

In the meantime, the EA docs / code should recommend a few things

1. Use `pandas_dtype(dtype)` to convert strings like `Period[D]` to the ExtensionDtype or numba dtype
2. For extension types, recommending `dtype.construct_array_type()._from_sequence` *may* make sense, but I'm not sure. That's supposed to take sequences of the scalar, so in theory it'll only work when you have to different ExtensionDtypes for the same scalar type. Not sure how common that'll be.

**Comment by jbrockmendel** [comment]
the ndarray/EA return type is now clear in the docstring.  Closing.
